app/app/Http/Controllers/ChangePasswordController.php

```diff
@@ -32,36 +33,25 @@ public function post(Request $request)
         if (BaseUser::isSignedIn())
         {
 			$user = BaseUser::getDbUser();
-			$validation_rules = array(
-				'current_password'     => 'required',
-				'new_password'         => 'required',
-				'password_confirm'     => 'required|same:new_password'
-			);
-			$validator = Validator::make(Input::all(), $validation_rules);
-			$failing = $validator->fails();
-			if (!$failing && !empty($user->password_hash))
-			{
-				if ( !BaseUser::authenticate($user->email, $request->input('current_password')) )
-				{
-					$validator->errors()->add('current_password', 'Current password not matched');
-					$failing = true;
-				}
+			$validationRules = [
+                'current_password' => 'required',
+                'new_password' => 'required',
+                'password_confirm' => 'required|same:new_password'
+            ];
+			$validator = Validator::make($request->all(), $validationRules);
+			if ($validator->fails()) {
+			    return redirect()->to('change-password')->withErrors($validator->errors())->withInput();
+            } else {
+			    if (Hash::check($request->get('current_password'), $user->password_hash)) {
```

Comment:
You don't get why we're asking for the current password despite being signed in?

This can explain it: https://security.stackexchange.com/questions/24291/why-do-we-ask-for-a-users-existing-password-when-changing-their-password

Reply:
It'll be a little less extra code.  No need for the "use Illuminate\Support\Facades\Hash" at the top.  It is also to keep the hash checks in a single place.  I admit these aren't huge reasons but there's no bigger reason to use Hash::check either so just keep using BaseUser::authenticate as I requested.